nemo/collections/asr/parts/submodules/rnnt_decoding.py

```diff
@@ -867,6 +905,54 @@ def _compute_offsets(
         offsets = list(filter(lambda offsets: offsets["char"][0] != rnnt_token, offsets))
         return offsets
 
+    @staticmethod
+    def _compute_offsets_tdt(hypothesis: Hypothesis, *args) -> List[Dict[str, Union[str, int]]]:
+        """
+        Utility method that calculates the indidual time indices where a token starts and ends.
+
+        Args:
+            hypothesis: A Hypothesis object that contains `text` field that holds the character / subword token
+                emitted at a specific time step considering predicted durations of the previous tokens.
+
+        Returns:
+
+        """
+        # Merge the results per token into a list of dictionaries
+        offsets = [
+            {"char": [t, -1], "start_offset": int(s), "end_offset": int(s + d)}
+            for t, s, d in zip(hypothesis.text[0], hypothesis.timestep, hypothesis.token_duration)
+        ]
+        return offsets
+
+    @staticmethod
+    def _refine_timestamps(
+        encoded_char_offsets: List[Dict[str, Union[str, int]]],
+        char_offsets: List[Dict[str, Union[str, int]]],
+        supported_punctuation: Set,
+    ) -> List[Dict[str, Union[str, int]]]:
+
+        ## no refinement for rnnt
+
+        return encoded_char_offsets, char_offsets
+
+    @staticmethod
+    def _refine_timestamps_tdt(
```

Comment:
we use it in `_get_segment_offsets` (maybe only partially, but I guess in the future it can be extended). if we want to compute `supported_punctuation` here, we need to pass `vocabulary` to decoding object anyway(which will also only be used for this case). any difference to pass `vocabulary` or `supported_punctuation` through decoding object ?